### DISCUSSION TITLE
Compact P2 KSF HMI Air over Waveshare RS485 to POE eth (B)

### DISCUSSION THREAD

**Discussion by annoia** [veista/nilan#159] [Q&A]
I have the Nilan heatpump as described in the title and am trying to connect the Waveshare bridge to the port (now located on top of the unit). However, I'm not having any luck getting any kind of signal, and I don't get the fabled blue light on the bridge when connecting it. The configuration also just gives me a sad "Failed to Connect", so I haven't reached the "not supported" stage yet.

The connection to the heat pump is thus:
![PXL_20241120_193437331](https://github.com/user-attachments/assets/5016009f-366f-42c0-b3d6-8e0c8ebdd88b)

And these wires go to the bridge:
![PXL_20241120_193511403](https://github.com/user-attachments/assets/3b9c2506-5631-4218-b0ab-a50f443731ef)

I have fiddled with the waveshare config, but hopefully it's all good now:
![image](https://github.com/user-attachments/assets/28d637fd-6570-4bb8-b57c-792ba7327cff)

My potato also took a picture of this note inside the unit:
![image](https://github.com/user-attachments/assets/f34532dd-04a9-4be2-aa20-c3aeba51d20f)

And finally to prove that the note isn't lying:
![image](https://github.com/user-attachments/assets/142839e6-455f-41ca-97fb-43741b8747e3)

Any idea where I am failing? I can connect the ethernet cable to the plug next to the modbus connector and get information via the Nilan online portal, but I don't have any luck extracting it locally.

**Comment by k1ssj6n0s** [comment]
Do you have a Nilan gateway device?
If so, where does the cable come from to the Modbus connector on the gateway?
Directly from the connector you indicated or from another box?

I had a similar problem recently. In my case, the cable coming from the connector on top of the device was not suitable to establish a Modbus connection, because between the connector and the Nilan gateway there was another CTS 602 controller: I had to connect to this controller with the Waveshare device, not through the connector on top of the device.

**Reply by annoia** [reply]
To my knowledge I do not have a Nilan gateway. I am willing to inquire Nilan about connectivity, but I wanted to ask here first, if I am doing something obviously wrong (It has been a journey).

**Reply by annoia** [reply]
I just asked Nilan who quickly replied that the cabling hasn't changed and the wiring is correct - if the connector has been connected. So I guess to move on I will have to take the box apart to ensure the connector is wired up internally.

**Reply by veista** [reply]
The wiring indeed seems to be correct. Have you done a continuity test on the cable? I recommend using end ferrules on your wire ends.
